packages/rum/src/boot/startRecording.spec.ts

```diff
@@ -61,7 +57,13 @@ describe('startRecording', () => {
           defaultPrivacyLevel: DefaultPrivacyLevel.ALLOW,
         })
         .beforeBuild(({ lifeCycle, configuration, viewContexts, sessionManager }) => {
-          const recording = startRecording(lifeCycle, configuration, sessionManager, viewContexts, worker!)
+          const httpRequest = createHttpRequest(configuration.sessionReplayEndpointBuilder, SEGMENT_BYTES_LIMIT)
+
+          const requestSendSpy = spyOn(httpRequest, 'send')
+          ;({ waitAsyncCalls: waitRequestSendCalls, expectNoExtraAsyncCall: expectNoExtraRequestSendCalls } =
+            collectAsyncCalls(requestSendSpy))
+
+          const recording = startRecording(lifeCycle, configuration, sessionManager, viewContexts, worker!, httpRequest)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏use `interceptRequests` instead of introducing a sixth parameter here

Reply:
🤔 it would not play well with the `collectAsyncCalls` logic though and `interceptRequests` only deal with sync case for now.
Unless I am missing something, I'd be in favor of keeping those separated for now and see how it will evolve with the retry logic before trying to merge those helpers.

Reply:
Alright 👌